### ISSUE TITLE
WMF no longer detected

### ISSUE THREAD

**Issue by antonio-rojas** [ImageMagick/ImageMagick#7230]
### ImageMagick version

7.1.1-30

### Operating system

Linux

### Operating system, version and so on

Arch Linux x86_64

### Description

WMF is no longer detected at build time. The config log shows:
```
configure:33925: checking for wmf_lite_create in -lwmflite
configure:33947: gcc -o conftest -I/usr/include/webp -I/usr/include/webp  -I/usr/include/cairo -I/usr/include/freetype2 -I/usr/include/libpng16 -I/usr/include/harfbuzz -I/usr/include/glib-2.0 -I/usr/lib/glib-2.0/include -I/usr/include/sysprof-6 -pthread -I/usr/include/pixman-1 -I/usr/include/librsvg-2.0 -I/usr/include/gdk-pixbuf-2.0 -I/usr/include/cairo -I/usr/include/glib-2.0 -I/usr/lib/glib-2.0/include -I/usr/include/libmount -I/usr/include/blkid -I/usr/include/sysprof-6 -pthread -I/usr/include/freetype2 -I/usr/include/libpng16 -I/usr/include/harfbuzz -I/usr/include/pixman-1 -I/usr/include/libraw -I/usr/include/libpng16 -I/usr/include/pango-1.0 -I/usr/include/cairo -I/usr/include/glib-2.0 -I/usr/lib/glib-2.0/include -I/usr/include/harfbuzz -I/usr/include/freetype2 -I/usr/include/libpng16 -I/usr/include/pixman-1 -I/usr/include/libmount -I/usr/include/blkid -I/usr/include/sysprof-6 -pthread -I/usr/include/fribidi  -I/usr/include/OpenEXR -pthread -I/usr/include/Imath  -I/usr/include/lqr-1 -I/usr/include/glib-2.0 -I/usr/lib/glib-2.0/include -I/usr/include/sysprof-6 -pthread -I/usr/include/openjpeg-2.5  -I/usr/include/webp -I/usr/include/libpng16 -I/usr/include/freetype2 -I/usr/include/glib-2.0 -I/usr/lib/glib-2.0/include -I/usr/include/sysprof-6 -pthread -I/usr/include/harfbuzz -I/usr/include/fribidi -I/usr/include/freetype2 -I/usr/include/libpng16 -I/usr/include/harfbuzz -I/usr/include/glib-2.0 -I/usr/lib/glib-2.0/include -I/usr/include/sysprof-6 -pthread -I/usr/include/freetype2 -I/usr/include/libpng16 -I/usr/include/harfbuzz -I/usr/include/glib-2.0 -I/usr/lib/glib-2.0/include -I/usr/include/sysprof-6 -pthread -pthread     -fopenmp -Wall -march=x86-64 -mtune=generic -O2 -pipe -fno-plt -fexceptions         -Wp,-D_FORTIFY_SOURCE=2 -Wformat -Werror=format-security         -fstack-clash-protection -fcf-protection -g -ffile-prefix-map=/build/imagemagick/src=/usr/src/debug/imagemagick -flto=auto -fexceptions -pthread -DMAGICKCORE_HDRI_ENABLE=1 -DMAGICKCORE_QUANTUM_DEPTH=16     -DMAGICKCORE_HDRI_ENABLE=1 -DMAGICKCORE_QUANTUM_DEPTH=16  -Wl,-O1 -Wl,--sort-common -Wl,--as-needed -Wl,-z,relro -Wl,-z,now          -Wl,-z,pack-relative-relocs -flto=auto  conftest.c -lwmflite  -ljpeg -ljbig -lltdl  -lz -lzip -lXext -lXt  -lSM -lICE -lX11  -lbz2 -lm  -lpthread -lbz2 -ljpeg >&5
conftest.c:153:1: error: unknown type name 'namespace'
  153 | namespace conftest {
      | ^~~~~~~~~
conftest.c:153:20: error: expected '=', ',', ';', 'asm' or '__attribute__' before '{' token
  153 | namespace conftest {
      |                    ^
conftest.c: In function 'main':
conftest.c:159:8: error: 'conftest' undeclared (first use in this function)
  159 | return conftest::wmf_lite_create ();
      |        ^~~~~~~~
conftest.c:159:8: note: each undeclared identifier is reported only once for each function it appears in
conftest.c:159:16: error: expected ';' before '::' token
  159 | return conftest::wmf_lite_create ();
      |                ^~
      |                ;
configure:33947: $? = 1
```

### Steps to Reproduce

Build 7.1.1-30 with `--with-wmf`

**Comment by kmilos** [comment]
FWIW, looks like the same happened when bumping the MSYS2 package:
```
checking for WMF support... 
  checking for libwmf/ipa.h... yes
  checking for wmf_lite_create in -lwmflite... no
  no -- some components failed test
  checking if WMF package is complete... no
```
-29 built w/ WMF fine previously.

**Comment by Biswa96** [comment]
* bad workaround: `export ac_cv_lib_wmflite_wmf_lite_create=yes`
* good workaround:

```diff
--- a/configure.ac
+++ b/configure.ac
@@ -2915,6 +2915,7 @@
     else
         AC_MSG_RESULT([no])
     fi
+    AC_LANG_POP([C++])
 fi
 
 AM_CONDITIONAL([UHDR_DELEGATE],[test "$have_uhdr" = 'yes'])
```

`AC_LANG_PUSH([C++])` was added in 4b85373fcd4de75118c0bb62266b66f8245906ec commit.

**Comment by Biswa96** [comment]
The configure script should fail if user provides `--with-xyz` options explicitly. That may cause issue with existing use cases where such silent failures are ignored.

**Comment by exekutive** [comment]
--withwmf fails to compile for me as well

I tried adding `AC_LANG_POP([C++])` to my config.ac file as shown above , but still fails.

[config.log](https://github.com/ImageMagick/ImageMagick/files/14916363/config.log)

IM 7.1.1-31
MacOS 14
ARM64

**Comment by Biswa96** [comment]
Did you run autoreconf after changing configure.ac?

**Comment by urban-warrior** [comment]
Thank you for reporting the issue. We have successfully reproduced it and are actively working on a patch to resolve it. You can expect this patch to be merged into the [main GIT branch](https://github.com/ImageMagick/ImageMagick), later today. As part of our commitment to quality, this fix will also be included in the upcoming [beta releases of ImageMagick](https://imagemagick.org/archive/beta/) by tomorrow. Your patience and feedback are greatly appreciated.
